internal/controller/replication.storage/volumereplication_controller.go

```diff
@@ -772,6 +773,17 @@ func setFailureCondition(instance *replicationv1alpha1.VolumeReplication, errMes
 	}
 }
 
+func setReplicationCondition(instance *replicationv1alpha1.VolumeReplication, dataSource, statusMessage string, replicationStatus proto.GetVolumeReplicationInfoResponse_Status) {
```

Comment:
it is a bit unclear what `dataSource` is (untyped string), a comment or specific type would be helpful. That counts for this function, but also for the others that are called from here. A comment for a function should not just repeat it's name with spaces between the words, but be useful for someone to understand how/why it can be used.